stdlib/public/core/Builtin.swift

```diff
@@ -584,3 +584,42 @@ internal func _unsafeDowncastToAnyObject(fromAny any: Any) -> AnyObject {
   return any as! AnyObject
 #endif
 }
+
+// Game the SIL diagnostic pipeline by inlining this into the transparent
+// definitions below after the stdlib's diagnostic passes run, so that the
+// `staticReport`s don't fire while building the standard library, but do
+// fire if they ever show up in code that uses the standard library.
+@inline(__always)
+public // internal with availability
+func _trueAfterDiagnostics() -> Builtin.Int1 {
+  return true._value
+}
+
+/// Return the dynamic type of a value.
+@_transparent
+@_semantics("typechecker.type(of:)")
+public func type<Type, Metatype>(of: Type) -> Metatype {
+  // This implementation is never used, since calls to `Swift.type(of:)` are
+  // resolved as a special case by the type checker.
+  Builtin.staticReport(_trueAfterDiagnostics(), true._value,
+    ("internal consistency error: 'type(of:)' operation failed to resolve"
+     as StaticString).utf8Start._rawValue)
+  Builtin.unreachable()
+}
+
+/// Return the dynamic type of a value.
+@_transparent
+@_semantics("typechecker.withoutActuallyEscaping")
+public func withoutActuallyEscaping<ClosureType, ResultType>(
```

Comment:
Did you actually mean to include this since there's no support for the semantic yet?